pkg/config/tlscfg/options.go

```diff
@@ -142,6 +142,28 @@ func (o *Options) ToOtelClientConfig() configtls.ClientConfig {
 	}
 }
 
+// ToOtelServerConfig provides a mapping between from Options to OTEL's TLS Server Configuration.
+func (o *Options) ToOtelServerConfig() configtls.ServerConfig {
+	cfg := configtls.ServerConfig{
```

Comment:
what about translating Enabled to Insecure?

Reply:
@yurishkuro The `Insecure` field doesn't actually exist on `configtls.ServerConfig`, it only exists on `configtls.ClientConfig` (see https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/configtls.go#L112-L126). The `TLSSetting` field in `configgrpc.TLSSetting` and `confighttp.TLSSetting` is a pointer type so setting it to nil implies no TLS/ insecure (see https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configgrpc/configgrpc.go#L168-L171).

Reply:
I changed this method to return a pointer and return nil when `!o.Enabled`.